### ISSUE TITLE
Option to export page relationship data

### ISSUE THREAD

**Issue by mschrader15** [dvargas92495/generate-roam-site-action#16]
I don't find Roam's Graph Overview inside of Roam useful, but I could see it having navigation purposes on a personal blog with far fewer nodes.

I wrote some code to generate a [list of nodes and edges](https://github.com/mschrader15/personal-roam-website/blob/main/src/graph_data/test.json) to run after [generate-site.ts](https://github.com/dvargas92495/generate-roam-site-action/blob/main/src/generate-site.ts) has completed. 

Is this a feature that we see others wanting? If so, we could integrate the node-edge data generation into [generate-site.ts](https://github.com/dvargas92495/generate-roam-site-action/blob/main/src/generate-site.ts) itself to eliminate the file I/O.

It would obviously be up to the person with the website to determine if & how they want to display the data.

**Comment by caffo** [comment]
That's very interesting, Max. Do you have an example of how are you surfacing this data? One of the items we have in the pipeline is adding a variable to expose the page references, so it can be used anywhere in the user layout (like we have `$PAGE_NAME` now).

**Comment by Stvad** [comment]
#15 has also graph overview with d3js fyi. maybe worth extracting that one in a common package to be reused by both?

**Comment by mschrader15** [comment]
@caffo here is the code: https://github.com/mschrader15/generate-roam-site-action/blob/generate-graph/src/generate-graph.ts

Incorporating it into [generate-site.ts](https://github.com/dvargas92495/generate-roam-site-action/blob/main/src/generate-site.ts) would eliminate the need to read in & parse the HTML. It currently creates separate edges for multiple links on the same page, which is incorrect. 

The data is put into the [sigma.js](http://sigmajs.org/) format but personally, I think I will switch to d3js. 

This is the first Node.js & TypeScript project I have worked on so it's probably pretty ugly haha   

@Stvad can you link to that code??

**Comment by caffo** [comment]
I wanted to see where you used it. Is it this? https://maxschrader.io/graph.html 

Love the clock btw :)

**Comment by mschrader15** [comment]
Yep, that's it. Ideally, it would have functionality and render itself properly but I haven't put much time into it yet besides copying a sigma.js example.

Thanks! I am in the process of building a base for it - I should have some pictures up of its final form by the end of the week

**Comment by Stvad** [comment]
@mschrader15 https://github.com/mathieudutour/gatsby-digital-garden/blob/master/packages/gatsby-theme-garden/src/components/graph-visualisation.js
